File: solutions/go/cars-assemble/1/cars_assemble.go
// Package cars provides simple calculations for car production rates and costs.
package cars

const (
	// singleCarCost is the cost of producing one car.
	singleCarCost = 10000

	// tenCarCost is the discounted cost of producing 10 cars as a batch.
	tenCarCost = 95000
)

// CalculateWorkingCarsPerHour returns the number of successful cars produced per hour.
// successRate is expressed as a percentage (e.g. 90.0 means 90%).
func CalculateWorkingCarsPerHour(productionRate int, successRate float64) float64 {
	return float64(productionRate) * successRate / 100.0
}

// CalculateWorkingCarsPerMinute returns the whole number of successful cars
// produced per minute. Fractional cars are truncated.
func CalculateWorkingCarsPerMinute(productionRate int, successRate float64) int {
	return int(CalculateWorkingCarsPerHour(productionRate, successRate) / 60.0)
}

// CalculateCost returns the total production cost for the given number of cars.
// Groups of 10 cars are charged at a discounted rate.
func CalculateCost(carsCount int) uint {
	return (uint(carsCount)/10)*tenCarCost + (uint(carsCount)%10)*singleCarCost
}

File: solutions/csharp/cars-assemble/1/CarsAssemble.cs
static class AssemblyLine
{
    public static double SuccessRate(int speed)
    {
        if (speed <= 0) return 0.0;   // invalid or zero speed → no production
        if (speed <= 4) return 1.0;   // speeds 1–4 → perfect
        if (speed <= 8) return 0.9;   // speeds 5–8 → minor errors
        if (speed == 9) return 0.8;   // speed 9 → noticeable errors
        return 0.77;                  // 10 → messy
    }

    // cars per hour with error rate factored in
    public static double ProductionRatePerHour(int speed) =>
        speed * 221.0 * SuccessRate(speed);

    // whole cars per minute
    public static int WorkingItemsPerMinute(int speed) =>
        (int)(ProductionRatePerHour(speed) / 60.0);
}